src/main/java/com/team/buddyya/feed/service/BookmarkService.java

```diff
@@ -11,8 +21,34 @@
 public class BookmarkService {
 
     private final BookmarkRepository bookmarkRepository;
+    private final StudentRepository studentRepository;
+    private final FeedService feedService;
 
-    public boolean isBookmarkedByStudent(Long studentId, Long feedId) {
+    public boolean existsByStudentIdAndFeedId(Long studentId, Long feedId) {
         return bookmarkRepository.existsByStudentIdAndFeedId(studentId, feedId);
     }
+
+    private Bookmark findByStudentIdAndFeedId(Long studentId, Long feedId) {
+        return bookmarkRepository.findByStudentIdAndFeedId(studentId, feedId)
+                .orElseThrow(() -> new FeedException(FeedExceptionType.FEED_NOT_BOOKMARKED));
+    }
+
+    public BookmarkResponse toggleBookmark(StudentInfo studentInfo, Long feedId) {
+        Feed feed = feedService.findFeedByFeedId(feedId);
+        boolean isBookmarked = existsByStudentIdAndFeedId(studentInfo.id(), feedId);
+        if (isBookmarked) {
+            Bookmark bookmark = findByStudentIdAndFeedId(studentInfo.id(), feedId);
+            bookmarkRepository.delete(bookmark);
+            return BookmarkResponse.from(false);
+        }
+        Student student = studentRepository.findById(studentInfo.id())
+                .orElseThrow(() -> new StudentException(StudentExceptionType.STUDENT_NOT_FOUND));
+
+        Bookmark bookmark = Bookmark.builder()
```

Comment:
P2: 컨벤션 상 개행은 없어도 될 것 같아요!
```suggestion
        Student student = studentRepository.findById(studentInfo.id())
                .orElseThrow(() -> new StudentException(StudentExceptionType.STUDENT_NOT_FOUND));
        Bookmark bookmark = Bookmark.builder()
```

Reply:
네 수정하겠습니다!